src/elm/Profile/Contact.elm

```diff
@@ -0,0 +1,406 @@
+            , problems = model.errors
+            , translators = translators
+            }
+            |> Input.toHtml
+        ]
+
+
+viewProfileInput : Translators -> Model -> Html Msg
+viewProfileInput ({ t } as translators) model =
+    div [ class "w-full" ]
+        [ Input.init
+            { label = t "contact_form.username.label"
+            , id = model.id ++ "_contact_input"
+            , onInput = EnteredContactText
+            , disabled = False
+            , value = model.contact
+            , placeholder = Just (t "contact_form.username.placeholder")
+            , problems = model.errors
+            , translators = translators
+            }
+            |> Input.toHtml
+        ]
+
+
+
+-- UTILITIES
+
+
+usesPhone : ContactType -> Bool
+usesPhone contactType =
+    List.member contactType [ ContactType.Whatsapp, ContactType.Phone ]
+
+
+unwrap : Normalized -> Contact
+unwrap (Normalized contact) =
+    contact
+
+
+hasSameType : Normalized -> Normalized -> Bool
+hasSameType (Normalized contact1) (Normalized contact2) =
+    contact1.contactType == contact2.contactType
+
+
+updateType : ContactType -> Model -> Model
+updateType newType model =
+    { model | contactType = newType, errors = Nothing }
+
+
+addErrors : List String -> Model -> Model
+addErrors errors model =
+    case errors of
+        [] ->
+            { model | errors = Nothing }
+
+        err :: _ ->
+            { model | errors = Just [ err ] }
+
+
+
+-- NORMALIZING
+
+
+countryCode : Country -> String
+countryCode country =
+    case country of
+        Brazil ->
+            "+055"
+
+        CostaRica ->
+            "+506"
+
+        Ethiopia ->
+            "+251"
+
+        UnitedStates ->
+            "+001"
+
+
+normalize : Country -> Validate.Valid Model -> Normalized
+normalize country validatedContact =
+    let
+        { contactType, contact } =
+            Validate.fromValid validatedContact
+    in
+    Normalized
+        { contactType = contactType
+        , contact =
+            case contactType of
+                ContactType.Instagram ->
+                    "https://instagram.com/" ++ contact
+
+                ContactType.Phone ->
+                    String.join " " [ countryCode country, contact ]
+
+                ContactType.Telegram ->
+                    "https://t.me/" ++ contact
+
+                ContactType.Whatsapp ->
+                    String.join " " [ countryCode country, contact ]
+        }
+
+
+
+-- VALIDATING
+
+
+phoneRegex : Regex
+phoneRegex =
+    Regex.fromString "^\\+?\\(?[0-9]{3}\\)?[-\\s\\.]?\\(?[0-9]{2}?\\)?[-\\s\\.]?[0-9]{3,5}[-\\s\\.]?[0-9]{3,4}$"
+        |> Maybe.withDefault Regex.never
+
+
+validator : ContactType -> Translators -> Validate.Validator String Model
+validator contactType translators =
+    let
+        phoneValidations =
+            [ Validate.ifBlank .contact (translators.t "contact_validator.phone.blank")
```

Comment:
we could add more validations here, like username on telegram (they have a few rules about this) 

<img width="256" alt="Captura de Tela 2021-02-24 às 17 08 24" src="https://user-images.githubusercontent.com/1082127/109059334-e903b780-76c2-11eb-9b5a-5032e7c441c2.png">

Reply:
Ok, I'll look into both telegram and instagram username rules